packages/react-components/virtualizer/src/useVirtualizer.ts

```diff
@@ -0,0 +1,403 @@
+import { useIntersectionObserver } from './useIntersectionObserver';
+import type { ReactNode } from 'react';
+import { useMemo, useRef, useState } from 'react';
+import * as React from 'react';
+
+import type { VirtualizerProps, VirtualizerState } from './Virtualizer.types';
+import { resolveShorthand } from '@fluentui/react-utilities';
+
+export function useVirtualizer_unstable(props: React.PropsWithChildren<VirtualizerProps>): VirtualizerState {
+  const {
+    itemSize,
+    virtualizerLength,
+    children,
+    sizeOfChild,
+    bufferItems = Math.round(virtualizerLength / 4.0),
+    bufferSize = Math.floor(bufferItems / 2.0) * itemSize,
+    scrollViewRef,
+    isReversed = false,
+    isHorizontal = false,
+    onUpdateIndex,
+    onCalculateIndex,
+  } = props;
+
+  // Safe access array version of children
+  const childArray = useMemo(() => (Array.isArray(children) ? children : [children]), [children]);
+
+  // Tracks the initial item to start virtualizer at, -1 implies first render cycle
+  const [virtualizerStartIndex, setVirtualizerStartIndex] = useState<number>(-1);
+
+  // Store ref to before padding element
+  const beforeElementRef = useRef<Element | null>(null);
+
+  // Store ref to before padding element
+  const afterElementRef = useRef<Element | null>(null);
+
+  // We need to store an array to track dynamic sizes, we can use this to incrementally update changes
+  const childSizes = useRef<number[]>(new Array<number>(sizeOfChild ? childArray.length : 0));
+
+  /* We keep track of the progressive sizing/placement down the list,
+  this helps us skip re-calculations unless children/size changes */
+  const childProgressiveSizes = useRef<number[]>(new Array<number>(sizeOfChild ? childArray.length : 0));
+
+  const populateSizeArrays = React.useCallback(() => {
```

Comment:
There's no reason `populateSizeArrays` should be a callback, it's never passed to a react component